infracheck/checks/smtp_credentials_check.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/python3
+"""
+Verifies SMTP credentials by using smtplib module to log in at the server.
+Arguments are provided via environmental variables:
+SMTP_HOST
+SMTP_PORT
+SMTP_USER
+SMTP_PASSWORD
+"""
+
+import smtplib
+import os
+import enum
+import sys
+
+
+class Messages(enum.Enum):
+    SUCCESS = 'Success'
+    SMTP_CONNECTION_ERROR = 'Error connecting to the SMTP server'
+    HELO_OR_EHLO_ERROR = 'HELO or EHLO error'
+    AUTHENTICATION_ERROR = 'Authentication failed'
+    DISCONNECTION_ERROR = 'Server is disconnected'
+    UNKNOWN_ERROR = 'Unknown error'
+    UNKNOWN_SMTP_ERROR = 'Unknown SMTP error'
+    CONNECTION_REFUSED_ERROR = 'Connection refused'
+    AUTH_METHOD_NOT_SUPPORTED_BY_SERVER = 'AUTH command not supported by the server'
+
+
+class EnvKeys(enum.Enum):
+    HOST = 'SMTP_HOST'
+    PORT = 'SMTP_PORT'
+    USERNAME = 'SMTP_USER'
+    PASSWORD = 'SMTP_PASSWORD'
+
+
+class SMTPCheck():
+    def main(self, host: str, port: int, username: str, password: str) -> tuple:
```

Comment:
Instead of "tuple" you can return `Tuple[int, str]`

It requires an import in Python up to 3.9, in unreleased Python 3.10 devs improved this, but 3.10 is unreleased yet :stuck_out_tongue: 
```python
from typing import Tuple
```